Fix command isolator recovery

komuta opened the pull request:
Restore container context (ContainerConfig) saved on `prepare`.

komuta committed:
Fix command isolator recovery

Restore container context (ContainerConfig) saved on `prepare`.

Lqp1 commented on src/CommandIsolator.cpp:
```diff
@@ -82,13 +92,37 @@ CommandIsolatorProcess::CommandIsolatorProcess(
   } else {
     m_infos.put(containerId, containerConfig);
   }
+  saveContainerContext(containerId, containerConfig);
```
Should we try to do something if we fail here? (open question, I have no idea how we can recover from this :( )

komuta replied:
Good point. I think the discussion is open. I'm not sure we'd like to ensure recovery to the point of preventing a task from launching just because we couldn't save its context on disk. But on the other hand, if some bug is massively happening, you'd like to get a clear signal... Let's discuss this point with the others.

komuta committed:
Add mandatory name parameter for isolator

This will allow to namespace context file per isolator.
It can also be used later for logs.

komuta committed:
Create isolator context dir

komuta committed:
Remove container context file on cleanup

kamaradclimber commented on src/ConfigurationParser.cpp:
```diff
@@ -95,6 +97,9 @@ Configuration ConfigurationParser::parse(
 
   configuration.isDebugSet = getOrEmpty(p, DEBUG_KEY) == "true";
+
+  configuration.name = getOrEmpty(p, MODULE_NAME_KEY);
```
what happens if it is empty?

komuta replied:
The universe collapses 💥 
The string remains empty (which is not good because all isolators will write their context in the same file).
Quick failure is not a good idea I think, what could be nice would be to have a default name (based on isolator number for example) but I didn't manage to do this easily in ModuleFactory.cpp

kamaradclimber replied:
Let's discuss about fail fast approach then

komuta replied:
done

komuta committed:
Fix tests

komuta committed:
Fix clang-format

Worst formatter ever :/

komuta committed:
Raise an error if module_name parameter is missing

komuta committed:
Don't run cleanup command if prepare was never called

This will prevent cleanup from failing because prepare was never called
on this isolator (if for example a previous isolator failed in prepare
stage) so context is not available.

komuta committed:
Add tests for recover feature

kamaradclimber commented on src/ConfigurationParser.cpp:
```diff
@@ -95,6 +97,12 @@ Configuration ConfigurationParser::parse(
+
+  configuration.name = getOrEmpty(p, MODULE_NAME_KEY);
+  if (configuration.name.empty())
+    throw std::runtime_error("At least on module is missing mandatory " +
```
error message is not clear to me. Maybe we could replace by: "Module name is missing for + MODULE_NAME_KEY

komuta replied:
`MODULE_NAME_KEY` is just the name of the parameter. What would be nice is to be able to point which module (isolator index for example) is concerned, but I wasn't able to extract this information (that's the whole point of naming explicitly modules). So to me we can't do much more that warn the user that (at least) one specified module doesn't have a name.

kamaradclimber replied:
I suggest the following message:

> "At least one module is missing '" + MODULE_NAME_KEY + "' parameter."

komuta committed:
Adjust missing module_name parameter error message
